assets/js/googlesitekit/data/create-settings-store.js

```diff
@@ -0,0 +1,392 @@
+				return actions.receiveSaveSettings( savedValues );
+			} catch ( err ) {
+				// TODO: Implement an error handler store or some kind of centralized
+				// place for error dispatch...
+				return actions.receiveResetFailed();
+			}
+		},
+
+		/**
+		 * Dispatches an action that creates an HTTP request to save settings.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		fetchSaveSettings() {
+			return {
+				payload: {},
+				type: FETCH_SAVE_SETTINGS,
+			};
+		},
+
+		/**
+		 * Dispatches that settings were saved via the REST API.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @param {Array} values Settings with their values from the API.
+		 * @return {Object} Redux-style action.
+		 */
+		receiveSaveSettings( values ) {
+			invariant( values, 'values is required.' );
+
+			return {
+				payload: {},
+				type: RECEIVE_SAVE_SETTINGS,
+			};
+		},
+
+		/**
+		 * Dispatches an action signifying the `fetchSaveSettings` side-effect failed.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		receiveSaveSettingsFailed() {
+			return {
+				payload: {},
+				type: RECEIVE_SAVE_SETTINGS_FAILED,
+			};
+		},
+	};
+
+	const controls = {
+		[ FETCH_SETTINGS ]: () => {
+			return API.get( type, identifier, datapoint );
+		},
+		// TODO: How do we get the current values to be available here?
+		[ FETCH_SAVE_SETTINGS ]: ( values ) => {
+			return API.set( type, identifier, datapoint, values );
+		},
+	};
+
+	const reducer = ( state = INITIAL_STATE, action ) => {
+		switch ( action.type ) {
+			case SET_SETTINGS: {
+				const { values } = action.payload;
+
+				return {
+					...state,
+					settings: {
+						...( state.settings || {} ),
+						...values,
+					},
+				};
+			}
+
+			case FETCH_SETTINGS: {
+				return {
+					...state,
+					isFetchingSettings: true,
+				};
+			}
+
+			case RECEIVE_SETTINGS: {
+				const { values } = action.payload;
+
+				return {
+					...state,
+					isFetchingSettings: false,
+					savedSettings: {
+						...values,
+					},
+					settings: {
+						...values,
+						// In case settings were already changed, they should take precedence.
+						...( state.settings || {} ),
+					},
+				};
+			}
+
+			case RECEIVE_SETTINGS_FAILED: {
+				return {
+					...state,
+					isFetchingSettings: false,
+				};
+			}
+
+			case FETCH_SAVE_SETTINGS: {
+				return {
+					...state,
+					isFetchingSaveSettings: true,
+				};
+			}
+
+			case RECEIVE_SAVE_SETTINGS: {
+				const { values } = action.payload;
+
+				return {
+					...state,
+					isFetchingSaveSettings: false,
+					savedSettings: {
+						...values,
+					},
+					settings: {
+						...values,
+					},
+				};
+			}
+
+			case RECEIVE_SAVE_SETTINGS_FAILED: {
+				return {
+					...state,
+					isFetchingSaveSettings: false,
+				};
+			}
+
+			default: {
+				// Check if this action is for a sub-setting reducer.
```

Comment:
I'm not exactly sure how this code is intended to work, but I think it'll make more sense when there are some tests around it. I don't quite follow what this means yet, but that's okay 😄

Reply:
The code further down iterates through all the passed setting slugs (e.g. `accountID`, `propertyID` and so on in the case of Analytics module), and then automatically creates actions like `setAccountID` (with type `SET_ACCOUNT_ID`) and selectors like `getAccountID`. In order for those to work it also needs to modify the overall reducer via this `settingReducers` variable.

So here in the `default` case, the main reducer checks whether there's a reducer defined for the current action type (e.g. `SET_ACCOUNT_ID`).